pyg_lib/csrc/sampler/cpu/neighbor_kernel.cpp

```diff
@@ -301,6 +328,8 @@ sample(const at::Tensor& rowptr,
           sampler.uniform_sample(/*global_src_node=*/sampled_nodes[i],
                                  /*local_src_node=*/i, count, mapper, generator,
                                  /*out_global_dst_nodes=*/sampled_nodes);
+          if constexpr (distributed)
+            cumm_sum_sampled_nbrs_per_node.push_back(sampled_nodes.size());
```

Comment:
`num_sampled_nodes_per_hop` and `cmm_sum_sampled_nbrs_per_node` are quite different. `num_sampled_nodes_per_hop` refers to the number of all nodes sampled in a layer, however `cmm_sum_sampled_nbrs_per_node` saves information of how many neighbors were sampled by each src node. So for example if there will be seed size = 4, and num_neighbors=[2], then `num_sampled_nodes_per_hop ` = [4, 8] and `cmm_sum_sampled_nbrs_per_node`=[4, 6, 8, 10, 12]. Based on the `cmm_sum_sampled_nbrs_per_node` we could calculate `num_sampled_nodes_per_hop`. Otherwise, I don't think so.